src/operator/reduce.ts

```diff
@@ -81,15 +81,20 @@ export class ReduceOperator<T, R> implements Operator<T, R> {
  * @extends {Ignored}
  */
 export class ReduceSubscriber<T, R> extends Subscriber<T> {
-  acc: T | R;
-  hasValue: boolean = false;
+  private index: number = 0;
+  private acc: T | R;
+  private hasValue: boolean = false;
 
   constructor(destination: Subscriber<R>,
-              private accumulator: (acc: R, value: T) => R,
+              private accumulator: (acc: R, value: T, index?: number) => R,
               seed: R,
               private hasSeed: boolean) {
     super(destination);
     this.acc = seed;
+
+    if (!this.hasSeed) {
+      this.index++;
+    }
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

> currentIndex The index of the current element being processed in the array. Starts at index 0, if an initialValue is provided, and at index 1 otherwise.

I tried to follow same behavior to native reduce, where it says start from 0 if seed exists, 1 otherwise.